File: Roadmap/01 - OPERADORES Y ESTRUCTURAS DE CONTROL/java/adrs1166ma.java
public class adrs1166ma {
    public static void main(String[] args) {
        // Operando1 = a
        // Operando2 = b
        // Respuesta = r
        int a = 5, b = 3, r = 0;
        System.out.println("Numero 1: "+a+" |  Numero 2:"+b);

        // ----------------------------------------------
        System.out.println("/* -- Aritmeticos -- */");
        System.out.println("Suma: "+(a+b));
        System.out.println("Resta: "+(a-b));
        System.out.println("Multiplicacion: "+(a*b));
        System.out.println("Division: "+(a/b));
        System.out.println("Residuo: "+(a%b));

        // ----------------------------------------------
        System.out.println("/* -- Logicos -- */");
        boolean t = true;
        boolean f = false;
        System.out.println(" -- Y [AND] -- ");
        System.out.println("false && false = " + (f && f));
        System.out.println("false && true  = " + (f && t));
        System.out.println("true  && false = " + (t && f));
        System.out.println("true  && true  = " + (t && t));

        System.out.println(" -- O  [OR] -- ");
        System.out.println("false || false = " + (f || f));
        System.out.println("false || true  = " + (f || t));
        System.out.println("true  || false = " + (t || f));
        System.out.println("true  || true  = " + (t || t));

        System.out.println(" -- Negacion [NOT] -- ");
        System.out.println("!true  = " + (!t));
        System.out.println("!false = " + (!f));

        // ----------------------------------------------
        System.out.println("/* -- Comparacion -- */");
        System.out.println("a == b = " + (a == b));
        System.out.println("a != b = " + (a != b));
        System.out.println("a > b = " + (a > b));
        System.out.println("a < b = " + (a < b));
        System.out.println("a >= b = " + (a >= b));
        System.out.println("a <= b = " + (a <= b));


        // ----------------------------------------------
        System.out.println(" -- Asignacion -- ");
        System.out.println("Asignación: " + (a = b));
        System.out.println("Suma y asignación: " + (a += b));
        System.out.println("Resta y asignación: " + (a -= b));
        System.out.println("Multiplicación y asignación: " + (a *= b));
        System.out.println("División y asignación: " + (a /= b));
        System.out.println("Resto y asignación: " + (a %= b));
        System.out.println("AND y asignación: " + (a &= b));
        System.out.println("OR y asignación: " + (a |= b));
        System.out.println("XOR y asignación: " + (a ^= b));
        System.out.println("Desplazamiento a la izquierda y asignación: " + (a <<= b));
        System.out.println("Desplazamiento a la derecha y asignación: " + (a >>= b));
        System.out.println("Desplazamiento a la derecha sin signo y asignación: " + (a >>>= b));

        // ----------------------------------------------

        /* NO IDENTIDAD  |  NO PERTENCIA */

        // ----------------------------------------------

        System.out.println(" -- Bits -- ");
        System.out.println("5 & 2 = " + (a & b));
        System.out.println("5 | 2 = " + (a | b));
        System.out.println("5 ^ 2 = " + (a ^ b));
        System.out.println("~5 = " + (~a));
        System.out.println("5 << 2 = " + (a << 2));
        System.out.println("5 >> 2 = " + (a >> 2));
        System.out.println("5 >>> 2 = " + (a >>> 2));

        // ----------------------------------------------

        /*
         * Estructuras de control
         */

        System.out.println(" -- Ternarios -- ");
        int mayorEdad = 18;
        String msg = mayorEdad >= 18 ? "Eres mayor de edad" : "Eres menor de Edad";
        System.out.println(msg);

        System.out.println(" -- if | else -- ");
        boolean miCondicion = false;
        if (miCondicion == true) {
            System.out.println(" Mi condición es verdadera ");
        } else if (miCondicion == false) {
            System.out.println(" Mi condicion es reconocida ");
        } else {
            System.out.println(" Mi condicion es reconocida ");
        }

        System.out.println(" -- for -- ");
        for (int i = 0; i < 10; i++){
            System.out.println("i = "+i);
        }

        int i = 0;
        System.out.println(" -- while -- ");
        while (i < 10){
            System.out.println("i = "+i);
            i++;
        }

        System.out.println(" -- do while -- ");
        do {
            System.out.println("i = "+i);
            i++;
        } while (i < 5);

        System.out.println(" -- switch -- ");
        switch (a) {
            case 1:
                System.out.println("a es 1");
                break;
            case 2:
                System.out.println("a es 2");
                break;
            case 3:
                System.out.println("a es 3");
                break;
            default:
                System.out.println("a no es 1, 2 o 3");
                break;
        }

        System.out.println(" -- break -- ");
        for (int j=0;j<5;j++){
            if (j==3){
                break;
            }
            System.out.println("j = "+j);
        }

        System.out.println(" -- continue -- ");
        for (int j=0;j<5;j++){
            if (j==3){
                continue;
            }
            System.out.println("j = "+j);
        }

        try {
            int resultado = a/0;
            System.out.println("Resultado: "+resultado);
        } catch (ArithmeticException e){
            System.out.println("Error: Division por cero");
        }

        /*
        * Extra
        */

        for (int x=10;x<=55;x++){
            if (x==16){
                continue;
            } else if ((x%3)==0) {
                continue;
            } else {
                System.out.println(x);
            }
        }
    }
}

File: Roadmap/01 - OPERADORES Y ESTRUCTURAS DE CONTROL/javascript/adrs1166ma.js
/*
 * EJERCICIO:
 * - Crea ejemplos utilizando todos los tipos de operadores de tu lenguaje:
 *   Aritméticos, lógicos, de comparación, asignación, identidad, pertenencia, bits...
 *   (Ten en cuenta que cada lenguaje puede poseer unos diferentes)
 * - Utilizando las operaciones con operadores que tú quieras, crea ejemplos
 *   que representen todos los tipos de estructuras de control que existan
 *   en tu lenguaje:
 *   Condicionales, iterativas, excepciones...
 * - Debes hacer print por consola del resultado de todos los ejemplos.
 *
 * DIFICULTAD EXTRA (opcional):
 * Crea un programa que imprima por consola todos los números comprendidos
 * entre 10 y 55 (incluidos), pares, y que no son ni el 16 ni múltiplos de 3.
 *
 * Seguro que al revisar detenidamente las posibilidades has descubierto algo nuevo.
 */

// 🔥 Aritméticos
let suma = 10 + 5;
let resta = 10 - 5;
let multiplicacion = 10 * 5;
let division = 10 / 5;
let modulo = 10 % 5;

// 🔥 Lógicos
let and = true && false; // false
let or = true || false; // true
let not = !true; // false

// 🔥 De comparación
let igual = 10 == 5; // false
let diferente = 10 != 5; // true
let mayorQue = 10 > 5; // true
let menorQue = 10 < 5; // false
let mayorOIgualQue = 10 >= 5; // true
let menorOIgualQue = 10 <= 5; // false

// 🔥 Asignación
let asignacion = 10;
asignacion += 5; // 10 + 5 = 15
asignacion -= 5; // 10 - 5 = 5
asignacion *= 5; // 10 * 5 = 50
asignacion /= 5; // 10 / 5 = 2
asignacion %= 5; // 10 % 5 = 0

// 🔥 Identidad
let igualEstricto = 10 === 10; // true
let diferenteEstricto = 10 !== 10; // false

// 🔥 Pertenencia
let array = [1, 2, 3];
let pertenece = 2 in array; // true

// 🔥 Bits
let andBits = 10 & 5; // 0
// 1010 (10) AND 
// 0101 (5)
// ----
// 0000 (0) , Si ambos bits son 1, es = 1, sino es 0.

let orBits = 10 | 5; // 15
// 1010 (10) OR 
// 0101 (5)
// ----
// 1111 (15) , Si al menos uno de los bits es 1, es = 1, sino es 0.

let xorBits = 10 ^ 5; 
// 1010 (10) XOR
// 0101 (5)
// ----
// 1111 (15) , Si los bits son diferentes, es = 1, si son iguales es 0.

let notBits = ~10; 
// 0000 1010 NOT(10)
// ---------
// 1111 0101 (-11), 0 se convierte en 1 y 1 se convierte en 0. 

let desplazamientoIzquierda = 10 << 2; // 40
// 0000 1010 (10) << 2 
// ---------
// 0010 1000 (40) , Desplaza los bits del número hacia la izquierda, y deja un 0

let desplazamientoDerecha = 10 >> 2; // 2
// 0000 1010 (10) >> 2
// ---------
// 0000 0010 (2) , Desplaza los bits del número hacia la derecha, los de la derecha se descartan.
desplazamientoDerecha = -10 >> 2; // -3
// 1111 0110 (-10) >> 2
// ---------
// 1111 1101 (-3) , Al desplazar a la derecha, este bit de signo se copia en las nuevas posiciones a la izquierda.
let desplazamientoDerechaSinSigno = 10 >>> 2; // 2
// 0000 1010 (10) >>> 2
// ---------
// 0000 0010 (2) , 

// Similar al desplazamiento a la derecha, 
// pero en lugar de copiar el bit de signo, 
// se añaden ceros por la izquierda.

desplazamientoDerechaSinSigno = -10 >>> 2; // 61
// 11110110 (-10) >>> 2
// ---------
// 00111101 (61)
// 
// En el desplazamiento a la derecha sin signo, 
// Se añaden ceros a la izquierda en lugar de copiar el bit de signo.


// 🔥 Condicionales
if (10 > 5) {
    console.log("10 es mayor que 5");
} else {
console.log("10 no es mayor que 5");
}

// 🔥 Iterativas
for (let i = 0; i < 10; i++) {
console.log(i);
}

let i = 0;
while (i < 10) {
console.log(i);
i++;
}

// 🔥 Excepciones
try {
let resultado = 10 / 0;
} catch (error) {
console.log("Se capturo error: división por cero");
}


// 🔥 Extra
for (let i = 10; i <= 55; i++) {
    //   (pares)  Y  (ni 16) Y  (ni multi de 3) O (el extremo 55)
    if ( (i%2==0) && (i!=16) && ( i % 3 !== 0 ) || (i == 55) ){
        console.log(i)
    }
}